examplemusig2.c

```diff
@@ -59,118 +74,104 @@ int main(void) {
 
     /**** Aggregate the public keys and batch commitments for each signer for all messages ****/
+    printf("\n______ Precomputation _____________________________________________________ \n");
     for (i = 0; i < NR_SIGNERS; i++) {
-        if (!musig2_signer_precomputation(&mcs_list[i].mc, serialized_pk_list, serialized_batch_list, NR_SIGNERS, NR_MESSAGES)) {
-            return -1;
-        }
+        res = musig2_signer_precomputation(&mcs_list[i].mc, serialized_pk_list, serialized_batch_list, NR_SIGNERS, NR_MESSAGES);
+        printf("  Signer %d: %s", i + 1, musig2_error_str(res));
     }
+    printf("--------------------------------------------------------------------------- \n\n");
 
 
-    printf("**** STATE 1 ************************************************************** \n");
+    /************** STATE 1 **************/
     /**** Signature ****/
-    printf("\n* Partial Signatures: \n");
-
-
+    printf("\nState 1: Partial Signatures _______________________________________________ \n");
     for (i = 0; i < NR_SIGNERS; i++) {
         /* Generate the partial signatures */
-        if (musig2_sign(&mcs_list[i], &mps1[i], MSG_1, MSG_1_LEN)){
-            printf(" S%d: ", i + 1);
-            print_hex(mps1[i].sig, MUSIG2_PARSIG_BYTES);
-        }
-        else {
-            printf("* Failed to generate signature for Signer %d.\n", i + 1);
-            return -1;
-        }
+        res = musig2_sign(&mcs_list[i], &mps1[i], MSG_1, MSG_1_LEN);
+        printf("  Signer %d: %s", i + 1, musig2_error_str(res));
     }
 
-
     /**** Aggregation ****/
-    printf("\n* Aggregate signature: \n");
-
-
-    if (musig2_aggregate_partial_sig(mps1, signature_1, NR_SIGNERS)){
-        printf(" S: ");
+    printf("\nState 1: Aggregate ________________________________________________________ \n");
+    res = musig2_aggregate_partial_sig(mps1, signature_1, NR_SIGNERS);
+    printf("  Signature %s", musig2_error_str(res));
+    if (res == MUSIG2_OK){
+        printf("  S .... ");
         print_hex(&signature_1[MUSIG2_AGGR_PUBKEY_BYTES], MUSIG2_PARSIG_BYTES);
-        printf(" R: ");
+        printf("  R .... ");
         print_hex(signature_1, MUSIG2_AGGR_PUBKEY_BYTES);
     }
     else {
-        printf("* Failed to aggregate signatures.\n");
         for (k = 0; k < NR_SIGNERS; k++) {
             musig2_context_sig_free(&mcs_list[k]);
         }
         return -1;
     }
 
     /**** Verification ****/
+    printf("\nState 1: Verification _____________________________________________________ \n");
+    res = musig2_prepare_verifier(&aggr_pk_1, serialized_pk_list, NR_SIGNERS);
+    printf("  Prepare   %s", musig2_error_str(res));
 
-    musig2_prepare_verifier(&aggr_pk_1, serialized_pk_list, NR_SIGNERS);
     /* Verify the aggregated signature with secp256k1_schnorrsig_verify */
-    if (musig2_verify(signature_1, MSG_1, MSG_1_LEN, &aggr_pk_1))
+    if (musig2_verify(signature_1, MSG_1, MSG_1_LEN, &aggr_pk_1)) {
```

Comment:
Any particular reason why don't we have `musig2_verify` return a MUSIG2_ERR? I think, in order to keep consistency across the lib, this function should also return the enum.